### ISSUE TITLE
Pasting a session ID into Resonite should open that session

### ISSUE THREAD

**Issue by Nytra** [Yellow-Dog-Man/Resonite-Issues#1120]
### Is your feature request related to a problem? Please describe.

It is sometimes difficult to join a session directly by ID. It can require getting another session orb then editing the URL inside, or using a custom made tool or mod. 

### Describe the solution you'd like

Upon pasting a session ID into Resonite, the session should be joined. Similar to how pasting a world record URL opens the world.

### Describe alternatives you've considered

Continuing to use the edit session orb method, a custom made tool, or a mod.

There is the option of using the session URL links but those are not currently working as per https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/160

### Additional Context

N/a

**Comment by Frooxius** [comment]
Where are you getting the session ID from? What format of session ID are you looking at?

By default, the SessionID should be just a GUID, which the system has no way of knowing that it represents a session ID. It could be RecordID too. It could be user. It could be anything.

**Comment by Nytra** [comment]
@Frooxius 

I was thinking of ones that you can get from the session orb by clicking "Get Session Orb" button in the dash, so like:

res-steam://...
lnl-nat://...

and ones you can construct like:

ressession:///...

and possibly ones just starting with "S-" like from the headless config files, although that would maybe be harder to detect.

**Comment by theR4K** [comment]
https://github.com/maksim789456/SessionUrlJoin

**Comment by ProbablePrime** [comment]
I intend to add this as a part of #1295 and #2260 .

I am a little concerned though as pasting something like this in could be alarming. So I'm thinking of adding a setting that when enabled will show a confirmation step before importing items from the clipboard.

It would look something like this:
![Image](https://github.com/user-attachments/assets/0d0cf7ae-4da3-41e1-8c98-08950c4a6185)

The settings to me seem quite powerful and privacy orientated.

**Comment by FoxoBread** [comment]
Clipboard confirmation would be highly appreciated.
I've had a few mishaps with a rogue paste.

**Comment by Banane9** [comment]
> I am a little concerned though as pasting something like this in could be alarming with an instant join. So I'm thinking of adding a setting that when enabled will show a confirmation step before importing items from the clipboard.

Agree on the confirmation for joining a world and other immediate actions like text / url. For images and videos, I suppose the import dialog is already essentially a confirmation dialog 🤔

**Comment by art0007i** [comment]
I think pasting a session url/id should maybe spawn a session orb instead? Though I don't know what should happen in cases where you don't have spawning items permissions.

**Comment by ProbablePrime** [comment]
Hmm an Orb is actually a good idea too. Hm.

I do think we're generally moving away from orbs though, or at least that's my memory of them from a random Discord conversation. I cannot remember the details.

**Comment by art0007i** [comment]
I see, but I think if orbs are going to be a thing of the past, they should get a replacement thing that serves the same purpose of being able to share a session with other people in the same world, and then everything that spawns session orb would spawn new thing instead, in which case this option is still valid.

**Comment by ProbablePrime** [comment]
I dont know either way.

I'll speak to Froox when I can.

**Comment by shiftyscales** [comment]
I'd consider the import dialogue sufficiently distinct in this case, @Banane9 - as you had noted, there are some content types which do not have an import dialogue at all as they do not have options to select on import.

Per Prime's settings mockup above- it is possible that we can also separate out by the type of content anyway meaning that users could selectively turn off the confirmation for asset types which already have an import dialogue prompt if they desired.

**Comment by Frooxius** [comment]
I don't think the explicit confirmation is necessary when pasting things, because the import dialog already serves that purpose - it tells you you're trying to import something.

The problem is that not everything has one right now, so some things are effectively without the confirmation and they just happen instantly. Best approach might be to just add a dialog for the things that are missing them.

This could be done for stuff like session URL too. We can even offer multiple choices there, so the dialog goes something like:
- Join Session
- Import as Session Orb

For moving away from Orbs - we do want to move away from them as the specific representation of worlds/sessions, but they'll be replaced by a more modern equivalent - e.g. like the Video or Audio players are.

**Comment by Banane9** [comment]
> we do want to move away from them as the specific representation of worlds/sessions, but they'll be replaced by a more modern equivalent - e.g. like the Video or Audio players are.

On that note, I do like the orbs. They have something kinda cute about them with the wobble when poked and it's nice to have the preview built right into them - any replacement should not miss out on that, I think.

**Comment by Frooxius** [comment]
No hard decision was made on that yet. They're likely going to follow the EntityInterface style, so you could replace the official ones with custom ones if you prefer.

However let's keep that discussion for that out of this thread, it's getting a bit off topic for the main one. The main point was that we can keep it as Session Orb now, since it'll be replaced with functional equivalent at some point and we'd just swap over. The details of that replacement are not relevant for this particular issue.

**Comment by Banane9** [comment]
> The details of that replacement are not relevant for this particular issue.

Sure, just a little bit of feedback for the future consideration :D

Although now that got me thinking of a session orb that turns into a gloopy when "opened"

**Comment by Frooxius** [comment]
Let's keep it on topic for this please.

**Comment by ProbablePrime** [comment]
> Best approach might be to just add a dialog for the things that are missing them.

Agree, will see what I can do to all of the ones that are applicable.

**Comment by ProbablePrime** [comment]
For a bunch of other issues:
- https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1295
- https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2260
- https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/1120

I'm experimenting with overhauling, all the import processes for urls  and routing them through the universal importer. 

It turns out, pasting a resrec from the clipboard will already work and open the world. So It made sense to add support for sessions there too.

I'm routing protocol links, that are opened from the browser, (see go.resonite.com), into the UniversalImporter as well.

Then I just need to add Session URL, LNL Urls, Steam URLs, etc  support to the UniversalImporter.

With this I'm also adding this option enum, which will have an option for both sessions and worlds: ![Image](https://github.com/user-attachments/assets/aa7d1d7c-1c8b-4008-8dfb-3b5de56fd0d3)

I'll also be adding an **optional confirmation step**(settings driven), for when you import the following items as they don't have a confirmation step. The confirmation step will default to off, But these can be turned on for privacy minded users, and I think its important to have the options.

1. resrec (records), items etc. (and worlds)
2. Sessions
3. Urls
4. Documents
5. Raw Files
6. Text, **when it is not being pasted into an active text field**

**Comment by ProbablePrime** [comment]
That's the first half of my previous message implemented :D  https://youtu.be/e8Wxe7UCcA8

**Comment by phil-flip** [comment]
@ProbablePrime 
Quick question: How does the helper program call Resonite? I am curious, because on Linux this might not work as expected, now that there is no native Linux version anymore.
Does it account that it runs on a Linux system and instead calls steam to start/call the game in proton?

**Comment by ProbablePrime** [comment]
The helper program does not support linux currently, because we do not officially support linux.

Its likely we can make proton work, but that would have to be covered after the initial release.

**Comment by phil-flip** [comment]
> The helper program does not support linux currently, because we do not officially support linux.

I would still like to know, how the running game gets the call to a session/world.

**Comment by Banane9** [comment]
At least for initially launching the game, wouldn't it be possible to simply forward to a parametrized steam application url?

**Comment by ProbablePrime** [comment]
> > The helper program does not support linux currently, because we do not officially support linux.
> 
> I would still like to know, how the running game gets the call to a session/world.

---
> At least for initially launching the game, wouldn't it be possible to simply forward to a parametrized steam application url?

You'll have to wait for the patch notes, sorry.

**Comment by ProbablePrime** [comment]
This was also added in: 2025.1.14.6.

Linux support should be another issue.

Thank you!
